sample-implementation.js

```diff
@@ -1,120 +0,0 @@
-/**
- * DISCLAIMER
- *
- * This implementation does not have any safety checks or optimizations.
- * Instead, the code is highly readable.
- */
-
-import _ from 'lodash';
-
-export function isComposable(obj) {
-  return _.isFunction(obj) && _.isFunction(obj.compose);
-}
-export function isDescriptor(obj) {
-  return _.isObject(obj);
-}
-function mixDescriptors(...args) {
-  const descr = {
-    methods: {},
-    properties: {},
-    deepProperties: {},
-    initializers: [],
-    staticProperties: {},
-    propertyDescriptors: {},
-    configuration: {},
-  };
-  _.forEach(args, (d) => {
-    if (!isDescriptor(d)) {
-      return;
-    }
-    _.assign(descr.methods, d.methods);
-    _.assign(descr.properties, d.properties);
-    _.merge(descr.deepProperties, d.deepProperties);
-    descr.initializers = descr.initializers.concat(d.initializers);
-    _.assign(descr.staticProperties, d.staticProperties);
-    _.merge(descr.propertyDescriptors, d.propertyDescriptors);
-    _.merge(descr.configuration, d.configuration);
-  });
-  return descr;
-}
-
-// Internal function. Used to create composable factories.
-function stamp(descriptor) {
```

Comment:
Haven't started the stamp implementation in the revision yet, so obviously we'll still need something like this. =)